crates/edr_provider/src/test_utils.rs

```diff
@@ -186,6 +186,95 @@ pub async fn run_full_block(url: String, block_number: u64, chain_id: u64) -> an
     let mined_block = builder.finalize(&mut state, rewards)?;
 
     let mined_header = mined_block.block.header();
+    for (expected, actual) in replay_block
+        .transaction_receipts()?
+        .into_iter()
+        .zip(mined_block.block.transaction_receipts().iter())
+    {
+        debug_assert_eq!(
```

Comment:
Tentative change request: I'm wondering if it would be better to use normal assert here in case the tests are ran in release mode.

Reply:
This is explicitly using debug asserts such that they are not run in release mode.

For this reason tests should never be run in release mode, but instead you should enable optimisations for the test profile

Reply:
We discussed in person, but for posterity: The `debug_assert`s are there for debugging purposes. They provide additional information for when the `assert` fails, specifically in the case of unequal `receipt_root`s.